src/core/utils/qfieldcloudutils.cpp

```diff
@@ -86,3 +88,145 @@ const QVariant QFieldCloudUtils::projectSetting( const QString &projectId, const
   const QString projectPrefix = QStringLiteral( "QFieldCloud/projects/%1" ).arg( projectId );
   return settings.value( QStringLiteral( "%1/%2" ).arg( projectPrefix, setting ), defaultValue );
 }
+
+const QMultiMap<QString, QString> QFieldCloudUtils::getPendingAttachments()
+{
+  QMultiMap<QString, QString> files;
+
+  QLockFile attachmentsLock( QStringLiteral( "%1/attachments.lock" ).arg( QFieldCloudUtils::localCloudDirectory() ) );
+  if ( attachmentsLock.tryLock( 10000 ) )
+  {
+    QFile attachmentsFile( QStringLiteral( "%1/attachments.csv" ).arg( QFieldCloudUtils::localCloudDirectory() ) );
+    QFileInfo fi( attachmentsFile );
+    if ( !fi.exists() || fi.size() == 0 )
+      return std::move( files );
```

Comment:
Don't you need to unlock when doing an early return?

Reply:
Actually I think I should remove the unlock altogether, I assume it auto unlocks in the class destructor kicks in when we scope out of the function

Reply:
You are right https://doc.qt.io/qt-6/qlockfile.html#dtor.QLockFile